### ISSUE TITLE
A/B Test: Turbo Mode Rebrand

### ISSUE THREAD

**Issue by athomasmoz** [mozilla-mobile/firefox-tv#2689]
### Why/User Benefit/User Problem

* User interviews showed that the "turbo mode" name did not reflect the added privacy from this feature.
* Better align with other Firefox branding & messaging

### Acceptance Criteria (how do I know when I’m done?)

* Turbo mode has been updated to Enhanced Tracking Protection
  * Onboarding screen
  * Toolbar icon & description
  * Settings icon & description
* The changes are behind an A/B test ([consultation ticket](https://bugzilla.mozilla.org/show_bug.cgi?id=1574528))
  * X% audience in A
  * X% audience in B
  * Remaining users got the current experience
* There is a data science request to analyze the results
  * Success means there isn't a significant drop in the number of users who keep turbo mode on during onboarding
  * More toggling of the turbo mode option is a good thing, but the change can still be a success without it
* The new changes should only be shown to EN users

### Design

Onboarding screen:

* Screenshot: https://drive.google.com/file/d/1HoQQtap4p4tB3CK8L6wfPhIQqcWEEtpE/view?usp=sharing
* Copy:
**Enhanced Tracking Protection**
Protect your privacy while you browse. Block invisible content that tracks the site you visit and profiles you. Blocking some of this content may make pages load faster.

Toolbar:

* Icon:
    * ON: https://drive.google.com/file/d/1HqzFEkje0PSmpkS_4yYHrTYQvotX866s/view?usp=sharing
    * OFF: https://drive.google.com/file/d/1Ht2zTXc189aSx6TW-Vmrjz9Ba8Vxwega/view?usp=sharing
* Description:
    * ON: Enhanced Tracking Protection ON
    * OFF: Enhanced Tracking Protection OFF

**Comment by Baron-Severin** [comment]
@brampitoyo the current Turbo onboarding buttons say "Keep Turbo Mode enabled" and "Turn Turbo Mode off".  What should their new copy be?

**Comment by brampitoyo** [comment]
@Baron-Severin I missed this. Thanks for pointing it out!

Enhanced Tracking Protection is quite a bit longer than Turbo Mode.

If we say “Keep Enhanced Tracking Protection enabled” that might make the button too long – too wordy to read. Let’s rename this option to “Enable Enhanced Tracking Protection”.

The other option can be renamed to “Turn Enhanced Tracking Protection off”.

**Comment by Baron-Severin** [comment]
Hm, sorry to resurrect this problem, but the buttons look a bit strange as is.  I'm not sure how we should approach this, @brampitoyo do you have any ideas?

![etc-onboarding](https://user-images.githubusercontent.com/13170306/63967142-a0c5f380-ca51-11e9-8ec6-9e254f96761b.png)

**Comment by athomasmoz** [comment]
@Baron-Severin just a reminder that we may need to restrict the experiment to EN only, unless you were able to re-use translations of the strings from desktop. Sorry, I should have noted it in the original description

**Comment by Baron-Severin** [comment]
> @Baron-Severin just a reminder that we may need to restrict the experiment to EN only, unless you were able to re-use translations of the strings from desktop. Sorry, I should have noted it in the original description

Thanks for catching that.  I'll add that change in now.

**Comment by brampitoyo** [comment]
@Baron-Severin That does look out of place. Does the button have a `max-width` value that we have to adhere to? Can we make it wider dynamically?

I’m open to renaming the button copy:

> Enhanced Tracking Protection ON

and

> Turn off

**Comment by Baron-Severin** [comment]
@brampitoyo we can definitely widen them.  I think having buttons this large looks a little strange, but it's not too bad.  I think if one of them is going to have a long string, they probably both should.  I'll make a quick change and send you a screenshot for approval.

**Comment by brampitoyo** [comment]
@Baron-Severin The only change I’d make is to rewrite the secondary button label to “Turn off”. The “Enable” button doesn’t look the best, but it works for the time being. Thanks!

**Comment by Baron-Severin** [comment]
@brampitoyo I've made the change in https://github.com/mozilla-mobile/firefox-tv/pull/2791.  There's a screenshot included there, would you mind double checking to make sure it's what you intend?

**Comment by abodea** [comment]
@Baron-Severin can you help me turn this on?
I tried this **`adb shell am start -n org.mozilla.tv.firefox/org.mozilla.tv.firefox.MainActivity -a android.intent.action.MAIN --esa qaActiveExperiments TurboModeRebrand-2689:B`** but I couldn't activate the experiment.

**Comment by athomasmoz** [comment]
Update on this ticket: for the A/B test, we can turn it to 100% for the new naming and just keep an eye on it over time. We don't have the tracking to plan out an A/B test for it at this time. Since the goal behind the rebrand is to align across products, not to increase usage, we should be ok to start it at 100%

**Comment by Baron-Severin** [comment]
> @Baron-Severin can you help me turn this on?
> I tried this **`adb shell am start -n org.mozilla.tv.firefox/org.mozilla.tv.firefox.MainActivity -a android.intent.action.MAIN --esa qaActiveExperiments TurboModeRebrand-2689:B`** but I couldn't activate the experiment.

@abodea it looks like there's a race condition in the code that prevents that command from taking effect.  I'm looking into it now.

**Comment by Baron-Severin** [comment]
> > @Baron-Severin can you help me turn this on?
> > I tried this **`adb shell am start -n org.mozilla.tv.firefox/org.mozilla.tv.firefox.MainActivity -a android.intent.action.MAIN --esa qaActiveExperiments TurboModeRebrand-2689:B`** but I couldn't activate the experiment.
> 
> @abodea it looks like there's a race condition in the code that prevents that command from taking effect. I'm looking into it now, but unfortunately it will require a code change to fix.

Following up on this: we set experiments to use values from the command line early during our startup process, but Fretboard does not allow us to do this in a blocking manner.  That means that anything querying that state within the next few MS will get incorrect values.  I've opened https://github.com/mozilla-mobile/android-components/issues/4488 to request a blocking call.

As for testing @abodea, I don't think there's any way for us to go through the normal process for this one.  Devs will have to test by hardcoding the experiment.  Sorry about that, I didn't realize this race condition existed when I tagged the story.

**Comment by athomasmoz** [comment]
@Baron-Severin I just realized that we can't send this out to all users because we held off localization pending the A/B test. Can we turn it on for all EN and I'll kick off the localization process in a separate ticket?

**Comment by Baron-Severin** [comment]
> @Baron-Severin I just realized that we can't send this out to all users because we held off localization pending the A/B test. I've put next steps into a separate given all the changes since we initially wrote this one #2888

Only displaying to EN users was part of the original AC for this ticket, so it's hardcoded in.  We can release this and turn it on in Kinto, and it'll still only show up for EN users.

**Comment by Baron-Severin** [comment]
Following up here.  I misinterpreted some documentation on Fretboard, and it looks like while the call is non-blocking, it should still give the correct response.  I did some additional debugging on our end, and afaict we're doing a substantial amount of setup before receiving and processing the intent that updates Fretboard state.  This is true for several other experiments as well, but we've been insulated from any negative effects by our reactive architecture (i.e., if state changes twice, the app generally updates itself to the new value without trouble).  Our turbo mode experiment is not reactive, exposing this issue.

I don't think it's worth changing in this instance (it would be a substantial amount of work to fix), but we should keep it in mind for the future.
